pkg/snmp/traps/oid_resolver.go

```diff
@@ -0,0 +1,185 @@
+// exist in a single file (after the previous conflict resolution), meaning that we get the variable
+// metadata from that same file.
+type MultiFilesOIDResolver struct {
+	traps TrapSpec
+}
+
+// NewMultiFilesOIDResolver creates a new MultiFilesOIDResolver instance by loading json or yaml files
+// (optionnally gzipped) located in the directory snmp.d/traps_db/
+func NewMultiFilesOIDResolver() (*MultiFilesOIDResolver, error) {
+	oidResolver := &MultiFilesOIDResolver{traps: make(TrapSpec)}
+	confdPath := config.Datadog.GetString("confd_path")
+	trapsDBRoot := filepath.Join(confdPath, "snmp.d", "traps_db")
+	files, err := os.ReadDir(trapsDBRoot)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read dir `%s`: %v", trapsDBRoot, err)
+	}
+	fileNames := getSortedFileNames(files)
+	for _, fileName := range fileNames {
+		err := oidResolver.updateFromFile(filepath.Join(trapsDBRoot, fileName))
+		if err != nil {
+			log.Warnf("unable to load trap db file %s: %s", fileName, err)
+		}
+	}
+	return oidResolver, nil
+}
+
+// GetTrapMetadata returns TrapMetadata for a given trapOID
+func (or *MultiFilesOIDResolver) GetTrapMetadata(trapOID string) (TrapMetadata, error) {
+	trapOID = strings.TrimSuffix(NormalizeOID(trapOID), ".0")
+	trapData, ok := or.traps[trapOID]
+	if !ok {
+		return TrapMetadata{}, fmt.Errorf("trap OID %s is not defined", trapOID)
+	}
+	return trapData, nil
+}
+
+// GetVariableMetadata returns VariableMetadata for a given variableOID and trapOID.
+// The trapOID should not be needed in theory but the Datadog Agent allows to define multiple variable names for the
+// same OID as long as they are defined in different file. The trapOID is used to differentiate between these files.
+func (or *MultiFilesOIDResolver) GetVariableMetadata(trapOID string, varOID string) (VariableMetadata, error) {
+	trapOID = strings.TrimSuffix(NormalizeOID(trapOID), ".0")
+	varOID = strings.TrimSuffix(NormalizeOID(varOID), ".0")
+	trapData, ok := or.traps[trapOID]
+	if !ok {
+		return VariableMetadata{}, fmt.Errorf("trap OID %s is not defined", trapOID)
+	}
+	varData, ok := trapData.variableSpecPtr[varOID]
+	if !ok {
+		return VariableMetadata{}, fmt.Errorf("variable OID %s is not defined", varOID)
+	}
+	return varData, nil
+}
+
+func getSortedFileNames(files []fs.DirEntry) []string {
+	fileNames := make([]string, 0, len(files))
+	for _, file := range files {
+		if file.IsDir() {
+			log.Debugf("not loading traps data from path %s: file is directory", file.Name())
+			continue
+		}
+		fileNames = append(fileNames, file.Name())
+	}
+
+	// Sort files alphabetically but put first the one shipped with the agent
+	sort.Slice(fileNames, func(i, j int) bool {
+		fileNameI := strings.ToLower(fileNames[i])
+		fileNameJ := strings.ToLower(fileNames[j])
+		if strings.HasPrefix(fileNameI, ddTrapDBFileNamePrefix) {
+			return true
+		} else if strings.HasPrefix(fileNameJ, ddTrapDBFileNamePrefix) {
+			return false
+		}
```

Comment:
Seems a bit hacky/complex to understand, maybe we can remove `dd_traps_db` from `fileNames` and insert it after it is sorted alphabetically ?